cpp/src/slice2cpp/Gen.cpp

```diff
@@ -3254,26 +3200,27 @@ Slice::Gen::InterfaceVisitor::visitInterfaceDefStart(const InterfaceDefPtr& p)
       << " current) const override;";
         H << sp;
     H << nl << "/**";
-    H << nl << " * Obtains the Slice type ID corresponding to this class.";
+    H << nl << " * Obtains the Slice type ID corresponding to this interface.";
     H << nl << " * @return A fully-scoped type ID.";
     H << nl << " */";
     H << nl << "static const ::std::string& ice_staticId();";
 
-    string flatName = "iceC" + p->flattenedScope() + p->name() + "_ids";
-
-    C << sp;
-    C << nl << "bool" << nl << scoped.substr(2) << "::ice_isA(::std::string s, const "
-      << getUnqualified("::Ice::Current&", scope) << ") const";
-    C << sb;
-    C << nl << "return ::std::binary_search(" << flatName << ", " << flatName << " + " << ids.size() << ", s);";
-    C << eb;
-
     C << sp;
     C << nl << "::std::vector<::std::string>" << nl << scoped.substr(2) << "::ice_ids(const "
       << getUnqualified("::Ice::Current&", scope) << ") const";
     C << sb;
-    C << nl << "return ::std::vector<::std::string>(&" << flatName << "[0], &" << flatName << '[' << ids.size()
-      << "]);";
+
+    // These type IDs are sorted alphabetically.
+    C << nl << "static const ::std::vector<::std::string> allTypeIds = ";
+    C.spar("{ ");
+    for (auto typeId : p->ids())
```

Comment:
shouldn't this be `auto&` or `const auto&`?

Reply:
Fixed